modin/numpy/arr.py

```diff
@@ -1469,6 +1559,283 @@ def sum(
+
+    def _isinf(
+        self,
+        /,
+        out=None,
+        *,
+        where=True,
+        casting="same_kind",
+        order="K",
+        dtype=None,
+        subok=True,
+    ):
+        check_kwargs(where=where, casting=casting, order=order, subok=subok)
+        result = self._query_compiler._isinf()
+        return fix_dtypes_and_determine_return(result, self._ndim, dtype, out, where)
+
+    def _isnan(
+        self,
+        /,
+        out=None,
+        *,
+        where=True,
+        casting="same_kind",
+        order="K",
+        dtype=None,
+        subok=True,
+    ):
+        check_kwargs(where=where, casting=casting, order=order, subok=subok)
+        result = self._query_compiler.isna()
+        return fix_dtypes_and_determine_return(result, self._ndim, dtype, out, where)
+
+    def _isnat(
+        self,
+        /,
+        out=None,
+        *,
+        where=True,
+        casting="same_kind",
+        order="K",
+        dtype=None,
+        subok=True,
+    ):
+        check_kwargs(where=where, casting=casting, order=order, subok=subok)
+        result = self._query_compiler._isnat()
+        return fix_dtypes_and_determine_return(result, self._ndim, dtype, out, where)
+
+    def _isneginf(self, out=None):
+        result = self._query_compiler._isneginf()
+        return fix_dtypes_and_determine_return(result, self._ndim, out=out)
+
+    def _isposinf(self, out=None):
+        result = self._query_compiler._isposinf()
+        return fix_dtypes_and_determine_return(result, self._ndim, out=out)
+
+    def _iscomplex(self):
+        result = self._query_compiler._iscomplex()
+        return fix_dtypes_and_determine_return(result, self._ndim)
+
+    def _isreal(self):
+        result = self._query_compiler._isreal()
+        return fix_dtypes_and_determine_return(result, self._ndim)
+
+    def _logical_not(
+        self,
+        /,
+        out=None,
+        *,
+        where=True,
+        casting="same_kind",
+        order="K",
+        dtype=None,
+        subok=True,
+    ):
+        check_kwargs(where=where, casting=casting, order=order, subok=subok)
+        result = self._query_compiler._logical_not()
+        return fix_dtypes_and_determine_return(result, self._ndim, dtype, out, where)
+
+    def _logical_binop(
+        self, qc_method_name, x2, out, where, casting, order, dtype, subok
+    ):
+        check_kwargs(where=where, casting=casting, order=order, subok=subok)
+        if is_scalar(x2):
+            return fix_dtypes_and_determine_return(
+                getattr(self, qc_method_name)(x2),
+                self._ndim,
+                dtype,
+                out,
+                where,
+            )
+        caller, callee, new_ndim, kwargs = self._binary_op(x2)
+        # breakpoint()
+        if caller._query_compiler != self._query_compiler:
+            # In this case, we are doing an operation that looks like this 1D_object,op(2D_object).
```

Comment:
Are we allowed to swap like this? I think so, but am curious about short circuiting - if we did or with two numerical arrays, would the result be the same if we swapped?

Reply:
Would short-circuiting ever matter here? Also, is there a different way we can do this broadcasted operation without rewriting a bunch of code from the `_binary_op` method?

Reply:
Sorry - by short circuiting I mean logic short circuiting:
```

In [1]: import numpy as np

In [2]: arr = np.array([[1, 2], [3, 4], [5, 6]])

In [3]: arr1 = arr + 1

In [4]: np.logical_or(arr, arr1)
Out[4]:
array([[ True,  True],
       [ True,  True],
       [ True,  True]])
```
seems this isn't a problem with logical or.
(For context, 
```
In [1]: 5 or 7
Out[1]: 5
```
is the short circuiting I was concerned about).

Reply:
Right, I figured short-circuiting wouldn't matter since the result is always coerced into a boolean here, so we shouldn't have to worry about keeping the exact numeric values around.

Reply:
Sounds good!